### ISSUE TITLE
Mousewheel scrolling erratic in Chrome 102

### ISSUE THREAD

**Issue by maciejregula** [petyosi/react-virtuoso#675]
**Describe the bug**
Fast mousewheel scrolling freezes table scroll in Chrome.

**Reproduction**
Use https://codesandbox.io/ to illustrate the problem so that I can observe the issue on my side and make sure that a potential fix reliably addresses it.

**To Reproduce**
1. Go to demo page: https://virtuoso.dev/hello/
2. Moseover the table a start fast scrolling using mousewheel
3. Table scroll doesn't work properly and freezes.

**Expected behavior**
Scroll is fluent and doesn't freeze. 

The issue doesn't happen when you mouseover the scrollbar itself in the table and start scrolling using mousewheel. Only when the mouse cursor is over the table.

**Video**
See video
https://user-images.githubusercontent.com/72083355/171611911-314e98b2-faec-46f7-a94a-6fb6b07a530b.mp4

**Desktop**
 - OS: Windows 10 pro 64bit
 - Chrome 102.0.5005.63

**Comment by petyosi** [comment]
I can't replicate the problem you experience using Windows and the most standard Logitech mouse + scroll wheel. My only guess is that this is something hardware-specific, but there's not enough information here for me to guess what could that be. Can you test on another device or with different hardware?

Since I have a whole lot of issues who never get replies after my initial reply, I will close this. I can reopen once there's something actionable.

**Comment by williamstein** [comment]
Hi.  I'm using the latest Chrome and macOS, and have a mouse, so I wanted to try to reproduce this.  

You wrote:

> 1. Go to demo page: https://virtuoso.dev/hello/
> 2. Moseover the table a start fast scrolling using mousewheel
> 3. Table scroll doesn't work properly and freezes.

There are a couple of typos and spelling errors in step 2, so I want to be sure I'm doing the right thing.  In particular, when you say "the table", do you mean the infinite *list* on the right, just below the heading "List with data"?  It's not really what I would call a "table".  Or, do you instead mean that actual table at https://virtuoso.dev/hello-table/ ?  If so, which table (there are 2 of them there)?

In any case, I'm not able to reproduce the problem using a standard Apple USB mouse with a scroll wheel.  I'm using an Apple Silicon macOS 12.4, Chrome 102.0.5005.61.

Do you see the problem using 2-finger scroll on the trackpad in addition to the physical mouse wheel?

(Disclaimer: I'm not a virtuoso developer (yet?), just a superfan user and financial contributor.)

**Comment by gwhobbs** [comment]
OK, I've tried `will-change: transform` on the container element as suggested in https://github.com/TanStack/virtual/issues/292 without luck. Adding `pointer-events: none` solves the issue for me, but prevents interaction with rows.

**Comment by maciejregula** [comment]
@williamstein Did you watch my video :) ? The problem occurs in "List with data". However, it also occurs in "List with total count". Yesterday I upgraded my Chrome to the version 102.0.5005.63. Also my colleagues from my work experienced this issue after upgrading Chrome (Linux/Windows). We couldn't reproduce the issue on OSX.

**Comment by williamstein** [comment]
Thanks -- sorry for my non-careful reading of your message.  I just tried with upgraded Chrome on Windows and wasn't personally able to reproduce this using my mouse (an apple usb mouse with a scroll wheel).

**Comment by maciejregula** [comment]
@gwhobbs see "scrollerRef" prop: https://virtuoso.dev/virtuoso-api-reference/. 
Please try to use `will-change: transfrom`  as below:
```js
scrollerRef={(ref) => {
  if (ref) {
    (ref as HTMLElement).style.willChange = "transform";
  }
}
```
Above workaround solved the issue for me in Chrome 102.

**Comment by petyosi** [comment]
@maciejregula this workaround does not make sense for Virtuoso, it does not use CSS transformations. If it works for you (which, again, has no logical explanation), you can set it like this: https://codesandbox.io/s/sandpack-project-forked-2dms32?file=/App.js. 

Everyone in this thread - please help by narrowing down the problem further to hardware or any specific additional changes.

**Comment by maciejregula** [comment]
@petyosi it doesn't make sense but works for me. Myabe there is a bug in Chrome...

**Comment by petyosi** [comment]
Here's a similar problem reported: https://discuss.codemirror.net/t/scrolling-issues-on-latest-google-chrome/4444/3. The workaround suggested can be applied to virtuoso as well: https://github.com/codemirror/CodeMirror/commit/1c5868673faa3802faf261c7ed0241180b7f59ac

**Comment by socarlosb** [comment]
Hello @maciejregula, did you try this example https://virtuoso.dev/window-scrolling/ ? Is this the behavior that you were expecting from the start? 🤔

**Comment by maciejregula** [comment]
@socarlosb I don't experience any issues with scrolling in the example you provided.

**Comment by socarlosb** [comment]
> @socarlosb I don't experience any issues with scrolling in the example you provided.

Because of the `useWindowScroll` attribute. I believe that is what you are searching for 👍

**Comment by slemvs** [comment]
Hi all, i have the same problem on Windows, chrome 102 and simple logitech mouse with scroll wheel. In my app react-virtuoso have scroll freezes when scrolling. 
This solution works for me: 
1. Mark main react-virtuoso component with css custom class
![image](https://user-images.githubusercontent.com/17675314/172362307-3b7528da-0037-4b32-86f7-b9384d24fa6e.png)

2. In css file in this custom class add `will-change: transform`
![image](https://user-images.githubusercontent.com/17675314/172362391-1df6928e-ddcf-445e-8d56-a6d8e55c7d7a.png)

**Comment by MGMehdi** [comment]
Hi,
I experience this issues too. I tried with Microsoft Edge (Version 102.0.1245.39) on the Virtuoso web site
I try on different computer with different mouse (wireless or not)

It's like it's loosing track of the scroll and I need to wait a second before doing a second scroll.
If you move the mouse in the table while you scroll it's working

This issue does not occur on Firefox (Version 101.0.1)

**Comment by Isaac96SM** [comment]
Hi,

I also have this issue. Do we know if it's going to be fixed?

Thanks

**Comment by petyosi** [comment]
It's a bug in Chrome, which, unfortunately, I can't reproduce on my side. There are workarounds provided from people who are reproducing it in their environments. Nothing else I can do. I am locking this issue, not much can be done further.

**Comment by petyosi** [comment]
The `will-change` workaround is now on by default.

**Comment by petyosi** [comment]
Everyone, as far as I saw, it looks like this problem does not happen in the latest Chrome release. Please confirm, as the workaround comes with its own problems.

**Comment by nosferatu500** [comment]
I also have this issue.

Electron v19.0.8
Chromium v102.0.5005.148

React-Virtuoso v2.16.3

**Comment by maximiliancsuk** [comment]
> It's no longer a problem in chrome 103 as far as I tested.

Same for me, the problem disappeared with some chrome update in the recent weeks.

**Comment by nosferatu500** [comment]
Hmm....

Electron 20.0.0-beta.8
Chromium 104.0.5073.0

Same issue.

**Comment by nosferatu500** [comment]
Thanks for the link. Guess I'll wait for the next stable version of Electron. If the problem still appears in the new version of Electron, but is resolved in Chrome 104, I will report it to Electron repository.

**Comment by nosferatu500** [comment]
About my case: Nothing special with Electron. I just copied your example from [here](https://virtuoso.dev/hello/) and scrolling is not working as expected. It just stops scrolling as soon as I stop scrolling mouse wheel. It's just freezing as soon as I stop scrolling mouse wheel.

**Comment by nosferatu500** [comment]
This issue has finally been fixed in Electron 19.0.10
